src/components/routes/Account/FirebaseEmailHandler/EmailVerify.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react';
+import AudioLoading from 'shared/loading/AudioLoading';
+import { auth } from 'utils/firebase';
+import Button from 'shared/Button';
+import BeeLink from 'shared/BeeLink';
+import { FirebaseConsumer, FirebaseUserProps } from 'HOCs/FirebaseProvider';
+
+interface State {
+  isSubmitting: boolean;
+  hasError: boolean;
+  showVerifyEmailSuccess: boolean;
+  successMessage: string;
+  errorMessage: string;
+}
+
+interface Props {
+  oobCode: string; //firebase code from email
+}
+
+export default class EmailVerify extends React.Component<Props> {
+  readonly state: State = {
+    isSubmitting: true,
+    hasError: false,
+    showVerifyEmailSuccess: false,
+    errorMessage: '',
+    successMessage: '',
+  };
+
+  componentDidMount() {
+    const oobCode = this.props.oobCode;
+
+    if (!oobCode) {
+      alert('Error, no oobCode defined.');
+      return;
+    }
+
+    this.handleVerifyEmail(oobCode);
+  }
+
+  handleVerifyEmail(oobCode: string) {
+    auth
+      .applyActionCode(oobCode)
+      .then(() => {
+        this.setState({
+          isSubmitting: false,
+          showVerifyEmailSuccess: true,
+          successMessage: 'Thanks for verifying your email.',
+        });
+      })
+      .catch(err => {
+        console.error(err);
+        this.setState({
+          isSubmitting: false,
+          hasError: true,
+          errorMessage: 'There was an error verifying your email.',
+        });
+      });
+  }
+
+  renderError() {
+    return (
+      <>
+        <h2>Sorry, there was an error.</h2>
+        <p>{this.state.errorMessage}</p>
+        <p>
+          <BeeLink href="mailto:support@beenest.com">
+            <Button>Contact us for further help.</Button>
+          </BeeLink>
+        </p>
+      </>
+    );
+  }
+
+  renderVerifyEmailSuccess() {
+    return (
+      <>
+        <h2>{this.state.successMessage}</h2>
+
+        <FirebaseConsumer>
+          {({ loading, user, completedVerification }: FirebaseUserProps) => {
+            if (loading) {
+              return <AudioLoading />;
```

Comment:
does this need a height/width to stay consistent with the rest of our loaders?

Reply:
if we want it to be consistent, should the AudioLoading have a default value inside the component?

Reply:
it has a default of 40/40 but i think it depends on where you're using it and which screen size you're using it on